### PR TITLE
updated policy 37 BWMPS to reflect new evidence shared by EAG

### PR DESCRIPTION
**Pull request by a-chacko**
I've updated the the script policy_37.R to reflect new weight loss and weight regain based on new papers suggested by EAG

All changes are only in this script [policy_37.R](https://github.com/nestauk/ahl_obesity_blueprint/blob/14-update-bwmp-inputs/analysis/policy_analysis/new_policies/policy_37.R)

### CODE REVIEW

File: analysis/policy_analysis/new_policies/sensitivity_testing_policy_38.R [hunk 1]
```diff
@@ -257,10 +257,27 @@ run_single_iteration <- function(seed_value,
...
+    filter(bmi_5_class %in% c("morbidly obese")) %>%
+    summarise(class_3_obesity_prevalence_y5 = sum(freq))
+  
+  # Calculating class 3 prevalence for baseline:
+  class_3_baseline_prevalence <- post_df_adult %>%
+    count(bmi_class, wt = wt_int) %>%
+    mutate(freq = n/sum(n)*100) %>%
+    filter(bmi_class %in% c("morbidly obese")) %>%
+    summarise(baseline_class_3_obesity_prevalence = sum(freq))
+
```
Comment (moniquetan) [inline analysis/policy_analysis/new_policies/sensitivity_testing_policy_38.R:287]: All good here. Essentially replicates the two blocks above, but just for morbid obesity, rather than obesity and morbid obesity.

File: analysis/policy_analysis/new_policies/sensitivity_testing_policy_38.R [hunk 2]
```diff
@@ -257,10 +257,27 @@ run_single_iteration <- function(seed_value,
...
+    filter(bmi_class %in% c("morbidly obese")) %>%
+    summarise(baseline_class_3_obesity_prevalence = sum(freq))
+  
   # Calculate reduction
   reduction <- (baseline_prevalence$baseline_obesity_prevalence - 
                   bmi_prevalence$obesity_prevalence_y5)
   
+  reduction_class_3 <- (class_3_baseline_prevalence$baseline_class_3_obesity_prevalence -
+    class_3_bmi_prevalence$class_3_obesity_prevalence_y5)
+
```
Comment (moniquetan) [inline analysis/policy_analysis/new_policies/sensitivity_testing_policy_38.R:294]: OK too. Replicating what's above, but limited to class 3 obesity (instead of all obesity).

File: analysis/policy_analysis/new_policies/sensitivity_testing_policy_38.R [hunk 3]
```diff
@@ -275,6 +292,10 @@ run_single_iteration <- function(seed_value,
     year5_obesity = bmi_prevalence$obesity_prevalence_y5,
     reduction = reduction,
     relative_reduction = (reduction / baseline_prevalence$baseline_obesity_prevalence) * 100,
+    baseline_class_3 = class_3_baseline_prevalence$baseline_class_3_obesity_prevalence,
+    year5_class_3_obesity = class_3_bmi_prevalence$class_3_obesity_prevalence_y5,
+    reduction_class_3 = reduction_class_3,
+    relative_reduction_class_3 = (reduction_class_3/class_3_baseline_prevalence$baseline_class_3_obesity_prevalence) * 100,
     benefit = value_to_gov,
```
Comment (moniquetan) [inline analysis/policy_analysis/new_policies/sensitivity_testing_policy_38.R:313]: OK here too

File: analysis/policy_analysis/new_policies/sensitivity_testing_policy_38.R [hunk 4]
```diff
@@ -455,7 +476,11 @@ summary_stats <- sensitivity_results %>%
     min_relative_reduction = round(min(relative_reduction), 3),
     max_relative_reduction = round(max(relative_reduction), 3),
     mean_benefit = mean(benefit),
-    sd_benefit = sd(benefit)
+    sd_benefit = sd(benefit),
+    mean_class_3_rel_reduction = round(mean(relative_reduction_class_3), 3),
+    sd_class_3_rel_reduction = round(sd(relative_reduction_class_3), 3),
+    min_class_3_rel_reduction = round(min(relative_reduction_class_3), 3),
+    max_class_3_rel_reduction = round(max(relative_reduction_class_3), 3),
   )
```
Comment (moniquetan) [inline analysis/policy_analysis/new_policies/sensitivity_testing_policy_38.R:498]: all good here too

**Review by moniquetan — COMMENTED** [review]
All good with amending the Tirzepatide sensitivity testing to get the class 3 obesity number. See point-by-point approvals in comments below.

I haven't reviewed the other changes made in earlier commits yet (eg BWMPs) and only focussed on the  Tirzepatide sensitivity testing for this comment/review.

I will do a separate review for BWMP.

File: analysis/policy_analysis/new_policies/sensitivity_testing_policy_38.R (hunk 1 above)
Comment (a-chacko) [reply analysis/policy_analysis/new_policies/sensitivity_testing_policy_38.R:287]: Yup, that's right!

File: analysis/policy_analysis/new_policies/policy_35.R [hunk 5]
```diff
@@ -55,6 +53,12 @@ source(file = "models/adult_model_calorie.R")
 source(file = "models/child_model_calorie.R")
 source(file = "post_processing/post_processing.R")
 
+# constants:
+NDNS_ADJUSTMENT = 1.32
+OOH_CONTRIBUTION_TO_DIETS = 0.11
+NON_PACKAGED_FOOD_CONTRBUTION_TO_DIETS = 0.11
+BASELINE_FOPL_COVERAGE = 0.66
+COMPENSATION_EFFECT = 0.23
```
Comment (moniquetan) [inline analysis/policy_analysis/new_policies/policy_35.R:62]: Constants match the slide here https://docs.google.com/presentation/d/1vW_9kjWoYoe_xvqEdC6rq15G9TT9F_fVnDik3OEQf98/edit?slide=id.g371f70717f1_0_1044#slide=id.g371f70717f1_0_1044, which is correct

File: analysis/policy_analysis/new_policies/policy_35.R [hunk 6]
```diff
@@ -68,16 +72,18 @@ process_clean_save(file_path = "inputs/raw/hse_2019_eul_20211006.tab", nation =
...
-# Inputs to the model:
-# Effect size [A]: ﹣12.4 kcal
-# Population segment impacted by policy [B]: Adults with BMI ≥ 25
-# Compensation effect [C]: 23% of [A] = 2.85
-# Duration [D]: 5 years ~ 365 * 5 days
 
-# Based on [A] and [C], the intake change = effect size - compensation effect = -9.57 kcals
+# calculating the change in intake:
+effect_size = 27
+effect_size_after_ndns_adjustment = effect_size * NDNS_ADJUSTMENT
```
Comment (moniquetan) [inline analysis/policy_analysis/new_policies/policy_35.R:78]: correct, multiply by 1.32

File: analysis/policy_analysis/new_policies/policy_35.R [hunk 7]
```diff
@@ -68,16 +72,18 @@ process_clean_save(file_path = "inputs/raw/hse_2019_eul_20211006.tab", nation =
...
+effect_size_ooh_adjustment = effect_size_after_ndns_adjustment * (1 - OOH_CONTRIBUTION_TO_DIETS)
```
Comment (moniquetan) [inline analysis/policy_analysis/new_policies/policy_35.R:79]: correct, multiply by 1-0.11 =0.89

File: analysis/policy_analysis/new_policies/policy_35.R [hunk 8]
```diff
@@ -68,16 +72,18 @@ process_clean_save(file_path = "inputs/raw/hse_2019_eul_20211006.tab", nation =
...
+effect_size_non_packaged_food_adjustment = effect_size_ooh_adjustment * (1 - NON_PACKAGED_FOOD_CONTRBUTION_TO_DIETS)
```
Comment (moniquetan) [inline analysis/policy_analysis/new_policies/policy_35.R:80]: correct, multiply by 1-0.11=0.89

File: analysis/policy_analysis/new_policies/policy_35.R [hunk 9]
```diff
@@ -68,16 +72,18 @@ process_clean_save(file_path = "inputs/raw/hse_2019_eul_20211006.tab", nation =
...
+effect_size_baseline_coverage_adjustment = effect_size_non_packaged_food_adjustment * (1 - BASELINE_FOPL_COVERAGE)
```
Comment (moniquetan) [inline analysis/policy_analysis/new_policies/policy_35.R:81]: correct, multiply by 1-0.66=0.34

File: analysis/policy_analysis/new_policies/policy_35.R [hunk 10]
```diff
@@ -68,16 +72,18 @@ process_clean_save(file_path = "inputs/raw/hse_2019_eul_20211006.tab", nation =
...
+effect_size_post_compensation = effect_size_baseline_coverage_adjustment * (1 - COMPENSATION_EFFECT)
```
Comment (moniquetan) [inline analysis/policy_analysis/new_policies/policy_35.R:82]: correct, multiply by 1-0.23=0.77

**Review by moniquetan — COMMENTED** [review]
FOP review: reviewed the changes made since my last review (9 June), which consisted of adjusting the effect size around NDNS underestimation, OOH contribution, non-packaged food contribution, etc., as laid out in this slide https://docs.google.com/presentation/d/1vW_9kjWoYoe_xvqEdC6rq15G9TT9F_fVnDik3OEQf98/edit?slide=id.g371f70717f1_0_1044#slide=id.g371f70717f1_0_1044. 

I confirm that all adjustments have been applied correctly.

File: analysis/policy_analysis/new_policies/policy_36.R
Comment (moniquetan) [inline analysis/policy_analysis/new_policies/policy_36.R:1]: MUP: 
- The energy intake changes by alcohol consumption categories match those given in Table 4 here https://docs.google.com/document/d/1dWnjV9dU5eSV3GdUpn45eQzampI3e3iz4SRJCnmsGHs/edit?tab=t.0#heading=h.i0hc2zoax9, which is correct
- These different EI changes are then applied to different people in the population based on their alcohol intake. I assume that the levels in the `alcohol_overall` field have been used correctly.
- Line 117 is hardcoded, presumably due to `bmi_model` being a matrix rather than a df? The result may be correct but this handling is really risky and error-prone. 
   - Must do: double-check that the correct columns are selected via hard-coding
   - Should do: make necessary amends so that hardcoding his no longer necessary

Comment (a-chacko) [reply analysis/policy_analysis/new_policies/policy_36.R:1]: Hey Monique, Updates below:
**Must do:**
- I've checked and can confirm that the right columns are being chosen.

**Should do:**
- I'll add this as an issue on github to pick up later

File: analysis/policy_analysis/new_policies/policy_37.R [hunk 11]
```diff
@@ -10,37 +10,50 @@
...
+#   programme to estimate the number of people who will be offered the treatment
+#   each year
+#   Number of people enrolled = ~1.2 million people per year
 #     Allocated budget / Unit cost per year = £85 mil / £70 = 1,214,285
-#
+#   
 #   Allocated budget = £85 million per year from the policy specification
-# 
 #   Cost of delivering the programme: £70 per person
 #     Unit cost of treatment in 2015 = £51.45 per person [Source]
```
Comment (moniquetan) [inline analysis/policy_analysis/new_policies/policy_37.R]: [Should] add missing source

File: analysis/policy_analysis/new_policies/policy_37.R [hunk 12]
```diff
@@ -10,37 +10,50 @@
...
 #     Adjusting for inflation = Original price x (CPI in 2015 / CPI in base year)
```
Comment (moniquetan) [inline analysis/policy_analysis/new_policies/policy_37.R]: [Should] specify what the base year is + source of conversion factor

File: analysis/policy_analysis/new_policies/policy_37.R [hunk 13]
```diff
@@ -10,37 +10,50 @@
...
 #         51.45 x (135.4/100) = £70 per person per year
+#   However, it is unlikely that everyone offered the programme will take up 
+#   treatment We use stats Taylor et al. (2024) (c) to estimate the number of 
+#   people who enroll and the share of people who experience weight loss
+#   Number of people experiencing weight loss = 45% of those who enrolled
+#   And number of people experiencing weight loss of those who enrolled
+#           = 45% x 1,214,285 = 546428.6; rounding this to 546,429
```
Comment (moniquetan) [inline analysis/policy_analysis/new_policies/policy_37.R]: [Should] this reads as if we are taking 45% of 45% of 1,2 million. But actually we are only applying the 45% once, not twice.

File: analysis/policy_analysis/new_policies/policy_37.R [hunk 14]
```diff
@@ -58,11 +71,26 @@ source(file = "post_processing/post_processing.R")
...
+
+# calculating the number of people experiencing weight loss:
+share_experiencing_weight_loss = 0.45
+
+number_enrolled_experiencing_weight_loss = share_experiencing_weight_loss * number_of_people_offered
+
+
 # Constants
-NUMBER_OF_PEOPLE_PER_YEAR = 1200000
+NUMBER_OF_PEOPLE_EXPERINCING_WEIGHT_LOSS = number_enrolled_experiencing_weight_loss
```
Comment (moniquetan) [inline analysis/policy_analysis/new_policies/policy_37.R]: nice to have it this way instead of hard-coding 1.2 million!

File: analysis/policy_analysis/new_policies/policy_37.R [hunk 15]
```diff
@@ -58,11 +71,26 @@ source(file = "post_processing/post_processing.R")
...
 ENGLAND_ADULT_POPULATION = 44263393  # (a)
```
Comment (moniquetan) [inline analysis/policy_analysis/new_policies/policy_37.R]: [should] (when revising codebase): take population figures out of policy-specific scripts and centralise

File: analysis/policy_analysis/new_policies/policy_37.R [hunk 16]
```diff
@@ -247,7 +275,7 @@ df = select_intervention_sample(data = df,
 
 
 # Assign weight changes to individuals who were selected in the previous step
-# We apply a weight loss of 2.4 kgs for those receiving the treatment and an annual weight regain of 0.32 kg
+# We apply a weight loss of 3.5 kgs for those receiving the treatment and an annual weight regain of 0.46 kg
```
Comment (moniquetan) [inline analysis/policy_analysis/new_policies/policy_37.R:301]: [should] weight loss and weight regain values are wrong? doesn't affect the modelling as that's just comments, but can create confusion.

File: analysis/policy_analysis/new_policies/policy_37.R [hunk 17]
```diff
@@ -237,7 +265,7 @@ set.seed(370)
 
 # Selecting the intervention sample each year who will receive the intervention:
 df = select_intervention_sample(data = df,
-                                sample_size = NUMBER_OF_PEOPLE_PER_YEAR,
+                                sample_size = NUMBER_OF_PEOPLE_EXPERINCING_WEIGHT_LOSS,
```
Comment (moniquetan) [inline analysis/policy_analysis/new_policies/policy_37.R]: correct to assign weight changes (weight loss + regain) only to those experiencing weight loss

File: analysis/policy_analysis/new_policies/policy_37.R [hunk 18]
```diff
@@ -58,11 +71,26 @@ source(file = "post_processing/post_processing.R")
...
+
+
 # Constants
-NUMBER_OF_PEOPLE_PER_YEAR = 1200000
+NUMBER_OF_PEOPLE_EXPERINCING_WEIGHT_LOSS = number_enrolled_experiencing_weight_loss
 ENGLAND_ADULT_POPULATION = 44263393  # (a)
-WEIGHT_LOSS_ON_TREATMENT = 2.4
-WEIGHT_REGAIN_POST_TREATMENT = 0.32
+WEIGHT_LOSS_ON_TREATMENT = 3.9 # from rapid review
+WEIGHT_REGAIN_POST_TREATMENT = 0.32 # from rapid review
```
Comment (moniquetan) [inline analysis/policy_analysis/new_policies/policy_37.R:70]: Annoyingly can't comment on lines that were not modified as part of this PR, but the function `assign_weight_changes()` is correct in assigning weight regain only at year > 1 of treatment.  

[should] (in codebase revision): need to come up with naming convention to differentiate between year of modelling vs year of treatment.

**Review by moniquetan — COMMENTED** [review]
Hi @a-chacko ,

**BWMP impact code review**

A few 'should' comments to look into when revising the codebase. 

My only 'must' would be to give me the source of the cost pp of BWMP to ensure we treat the correct number of people? According to https://docs.google.com/spreadsheets/d/1MnA8i5HKPDkJbjHMyAyGtQKe7kYIhHPcyx9zQo8jSjM/edit?gid=829546672#gid=829546672 it would be the Taylor 2024 paper but I can't seem to find the £51.45 value in there.



> Hi @a-chacko ,
> 
> BWMP impact code review
> 
> A few 'should' comments to look into when revising the codebase.
> 
> My only 'must' would be to give me the source of the cost pp of BWMP to ensure we treat the correct number of people? According to https://docs.google.com/spreadsheets/d/1MnA8i5HKPDkJbjHMyAyGtQKe7kYIhHPcyx9zQo8jSjM/edit?gid=829546672#gid=829546672 it would be the Taylor 2024 paper but I can't seem to find the £51.45 value in there.
> 

The costs are now being calculated differently and the steps are captured on this slide - [BWMPs costs](https://docs.google.com/presentation/d/1tJZ3USk3l6S7kO6vB2uKHFZ_oyfXlRDO9REeCpx1M8s/edit?slide=id.g3811b0359be_0_1066#slide=id.g3811b0359be_0_1066)

File: analysis/policy_analysis/new_policies/policy_38.R
Comment (cmbrennan002) [inline analysis/policy_analysis/new_policies/policy_38.R:1]: Question to discuss: Could we correct for this by rather that adding the weight to the 'weight_sum' column, we add 1/weight_sum, and then only stop drawing from that observation if the column reaches weight_sum. It would make the loop run a long time but might be feasible? If we think this is methodologically sound, can get some quick help to run this on EC2/ speed it up

File: analysis/policy_analysis/new_policies/policy_36.R
Comment (a-chacko) [reply analysis/policy_analysis/new_policies/policy_36.R:1]: For the should do - #19 has been created to address this

File: analysis/policy_analysis/new_policies/policy_37.R (hunk 18 above)
Comment (a-chacko) [reply analysis/policy_analysis/new_policies/policy_37.R:70]: Created an issue [#23](https://github.com/nestauk/ahl_obesity_blueprint/issues/23) to address should comment

**Comment by a-chacko** [comment]
Both branches in this PR has already been merged into dev. Not deleting the branch for now, just in case we need it later on.
